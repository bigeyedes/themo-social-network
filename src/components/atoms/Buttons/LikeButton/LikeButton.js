import styled, { css } from 'styled-components';
import like from './like-outline.svg';
import likeFilled from './like-active.svg';

const Like = styled.button`
	background: none;
	width: 25px;
	height: 25px;
	cursor: pointer;
	background-image: url(${like});
	background-repeat: no-repeat;
	background-size: 20px auto;
	background-position: center;
	border: none;
	&:hover {
		opacity: 0.7
	}

	${({ liked }) =>
    liked &&
    css`
      background-image: url(${likeFilled});
    `}
`

export default Like;