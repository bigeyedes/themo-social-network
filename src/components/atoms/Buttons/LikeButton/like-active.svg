<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 24.3.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Warstwa_1" xmlns:svgjs="http://svgjs.com/svgjs"
	 xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 512 512"
	 style="enable-background:new 0 0 512 512;" xml:space="preserve">
<style type="text/css">
	.st0{fill:#3975F5;}
</style>
<path class="st0" d="M475.7,72c-24.2-23.6-57.6-35.4-100.3-35.4c-11.8,0-23.9,2-36.1,6.1c-12.3,4.1-23.7,9.6-34.3,16.6
	c-10.6,7-19.7,13.5-27.3,19.6c-7.6,6.1-14.9,12.6-21.7,19.4c-6.9-6.9-14.1-13.3-21.7-19.4c-7.6-6.1-16.7-12.6-27.3-19.6
	c-10.6-7-22-12.5-34.3-16.6c-12.3-4.1-24.3-6.1-36.1-6.1C93.9,36.6,60.5,48.4,36.3,72C12.1,95.6,0,128.4,0,170.3
	c0,12.8,2.2,25.9,6.7,39.4c4.5,13.5,9.6,25,15.3,34.6c5.7,9.5,12.2,18.8,19.4,27.9c7.2,9,12.5,15.3,15.9,18.7c3.3,3.4,6,5.9,7.9,7.4
	l178.3,172c3.4,3.4,7.6,5.1,12.6,5.1c5,0,9.1-1.7,12.6-5.1l178-171.4c43.6-43.6,65.4-86.5,65.4-128.6C512,128.4,499.9,95.6,475.7,72
	z"/>
</svg>
