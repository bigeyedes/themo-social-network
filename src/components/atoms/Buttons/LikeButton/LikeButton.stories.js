import React from 'react'
import { storiesOf } from '@storybook/react';
import LikeButton from './LikeButton';

storiesOf('Atoms/Buttons', module)
	.add('Like button normal', () => <LikeButton></LikeButton>)
	.add('Like button active', () => <LikeButton liked></LikeButton>)
