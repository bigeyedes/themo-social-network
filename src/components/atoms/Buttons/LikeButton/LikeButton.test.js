import React from 'react'
import renderer from 'react-test-renderer';
import 'jest-styled-components'
import LikeButton from './LikeButton'
import likeOutline from './like-outline.svg'
import likeActive from './like-active.svg'

describe("Render like button component", () => {
	test("Matches the snapshot", () => {
		const tree = renderer.create(<LikeButton />).toJSON()
		expect(tree).toMatchSnapshot()
	});

	test("If is 0.7 opacity when hover", () => {
		const tree = renderer.create(<LikeButton />).toJSON();
		expect(tree).toHaveStyleRule('opacity', '0.7', {
			modifier: ':hover',
		})
	});

	test("If have standard background image", () => {
		const tree = renderer.create(<LikeButton />).toJSON();
		expect(tree).toHaveStyleRule('background-image', `url(${likeOutline})`)
	});

	test("If change background image active 'liked' is passed", () => {
		const tree = renderer.create(<LikeButton liked/>).toJSON();
		expect(tree).toHaveStyleRule('background-image', `url(${likeActive})`)
	});

	

});