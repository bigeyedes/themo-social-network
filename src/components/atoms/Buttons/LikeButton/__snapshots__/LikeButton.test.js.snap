// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render like button component Matches the snapshot 1`] = `
.c0 {
  background: none;
  width: 25px;
  height: 25px;
  cursor: pointer;
  background-image: url(like-outline.svg);
  background-repeat: no-repeat;
  background-size: 20px auto;
  background-position: center;
  border: none;
}

.c0:hover {
  opacity: 0.7;
}

<button
  className="c0"
/>
`;
